src/main/java/org/zalando/riptide/RestClient.java

```diff
@@ -20,133 +20,79 @@
  * ​⁣
  */
 
-import org.springframework.context.annotation.DependsOn;
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
-import org.springframework.http.MediaType;
-import org.springframework.http.client.ClientHttpRequest;
-import org.springframework.http.converter.HttpMessageConverter;
-import org.springframework.web.client.RestClientException;
 
-import javax.annotation.Nullable;
-import java.io.IOException;
 import java.net.URI;
-import java.util.List;
-import java.util.Optional;
 
-import static java.lang.String.format;
+abstract class RestClient {
 
-abstract class RestClient<R> {
-
-    public final Requester<R> get(final String urlTemplate, final Object... urlVariables) {
+    public final Requester get(final String urlTemplate, final Object... urlVariables) {
         return execute(HttpMethod.GET, urlTemplate, urlVariables);
     }
 
-    public final Requester<R> get(final URI url) {
+    public final Requester get(final URI url) {
```

Comment:
IMO all those methods should be covered in #106 - clarifying the usage of the `UriTemplateHandler`